Framework/DataObjects/src/CoordTransformDistance.cpp

```diff
@@ -28,36 +28,34 @@ namespace DataObjects {
  *dimensions that are considered when
  *        calculating distance.
  * @param outD :: # of output dimensions
+ * @param eigenvects :: eigenvectors of an ellipsoid (if used)
+ * @param eigenvals :: variances along each eigenvector (if used)
  * @return
  */
-CoordTransformDistance::CoordTransformDistance(const size_t inD,
-                                               const coord_t *center,
-                                               const bool *dimensionsUsed,
-                                               const size_t outD)
+CoordTransformDistance::CoordTransformDistance(
+    const size_t inD, const coord_t *center, const bool *dimensionsUsed,
+    const size_t outD, const std::vector<Kernel::V3D> eigenvects,
+    const std::vector<double> eigenvals)
     : CoordTransform(inD, outD) {
-  // Create and copy the arrays.
-  m_center = new coord_t[inD];
-  m_dimensionsUsed = new bool[inD];
+
+  m_maxEigenval = 0.0;
```

Comment:
Can I ask that we add
```c++
m_center.reserve(inD);
m_dimensionsUsed.reserve(inD);
```
before the loop?
It helps only allocated memory for the vectors once as we know the size of them.

Reply:
Thanks, I didn't know about this in C++!